lms/djangoapps/certificates/docs/decisions/003-cert-revocation-changes.rst

```diff
@@ -0,0 +1,58 @@
+##############################
+Certificate Revocation Changes
+##############################
+
+Status
+------
+Accepted
+
+Background
+----------
+The GeneratedCertificate model provides functionality to ``invalidate`` or
+mark a certificate as ``notpassing``.
+
+Up until recently, as part of revoking a course certificate, we would remove
+information associated with the course certificate record by setting the
+following model fields to an empty string:
+
+* ``verify_uuid``
+* ``download_uuid``
+* ``download_url``
+* ``grade``
+
+Additionally, we update the value of the certificate's ``status`` field
+depending on the operation:
+
+* The certificate's status changes to ``unavailable`` when invalidated.
+* The certificate's status changes to ``notpassing`` when marked as not passing.
+
+**When do we invalidate certificates?**
+
+* When removing a learner from the allowlist
+* By the ``edx-proctoring`` plugin, through the CertificateService, when a learner has been identified as having suspicious behavior during a proctored exam.
+* When adding a learner to the blocklist through the Instructor Dashboard
+* When performing certificate regeneration through the LMS Support Tools
+* When performing bulk certificate regeneration through the Instructor Dashboard
+
+**When do we mark certificates as not passing?**
+
+* When the ``COURSE_GRADE_NOW_FAILED`` signal is fired and iff the learner was previously passing the course.
+
+Decisions
+---------
+
+* We will no longer remove the value of the ``verify_uuid`` field during certificate revocation. The ``verify_uuid`` is used as part of the URL to the certificate itself (courses.edx.org/certificates/<verify_uuid>).
+
+* In "regular" and bulk certificate regeneration we will skip over the certificate invalidation step of regeneration.
```

Comment:
I should probably highlight that this is only if V2 of course certificates is enabled. Woops!